include/xtensor-blas/xlinalg.hpp

```diff
@@ -1514,6 +1514,174 @@ namespace linalg
+                                               xarray<value_type, XTENSOR_DEFAULT_LAYOUT>>;
+
+        result_type result;
+        auto&& a = view_eval<T::static_layout>(xa.derived_cast());
+        auto&& b = view_eval<O::static_layout>(xb.derived_cast());
+        if (naxes == 0)
+        {
+            // special case tensor outer product product
+            xt::dynamic_shape<std::size_t> result_shape(a.dimension() + b.dimension());
+            std::size_t j = 0;
+            for (std::size_t i = 0; i < a.dimension(); ++i)
+            {
+                result_shape[j++] = a.shape()[i];
+            }
+
+            for (std::size_t i = 0; i < b.dimension(); ++i)
+            {
+                result_shape[j++] = b.shape()[i];
+            }
+            // flatten a/b
+            auto vec_a = xt::ravel<T::static_layout>(a);
+            auto vec_b = xt::ravel<O::static_layout>(b);
+            // take the outer product of the two vectors
+            result = outer(vec_a, vec_b);
+            // reshape the result
+            result.reshape(result_shape);
+        }
+        else
+        {
+          // Sum of products over last n axes of A and the first n axis of b
+            XTENSOR_ASSERT(a.dimension() >= naxes);
+            XTENSOR_ASSERT(b.dimension() >= naxes);
+
+            auto as_it = a.shape().begin() + (a.dimension() - naxes);
+            auto bs_it = b.shape().begin();
+            std::size_t sum_len = 1;
+            for (std::size_t i = 0; i < naxes; ++i)
+            {
+                auto a_val = *as_it;
+                auto b_val = *bs_it;
+                // check for axes size match
+                if (a_val != b_val)
+                {
+                    throw std::runtime_error("Shape mismatch for sum");
+                }
+                else
+                {
+                    sum_len *= a_val;
+                }
+                ++as_it;
+                ++bs_it;
+            }
+            xt::dynamic_shape<std::size_t> result_shape;
+            std::size_t keep_a_len = 1;
+            for (auto it = a.shape().begin(); it != a.shape().begin() + (a.dimension() - naxes); ++it)
+            {
+                std::size_t len = *it;
+                keep_a_len *= len;
+                result_shape.push_back(len);
+            }
+            std::size_t keep_b_len = 1;
+            for (auto it = b.shape().begin() + naxes; it != b.shape().end(); ++it)
+            {
+                std::size_t len = *it;
+                keep_b_len *= len;
+                result_shape.push_back(len);
+            }
+            xarray<value_type, T::static_layout> a_mat = a;
+            a_mat.reshape({keep_a_len, sum_len});
+            xarray<value_type, O::static_layout> b_mat = b;
+            b_mat.reshape({sum_len, keep_b_len});
+            result = dot(a_mat, b_mat);
+            if(result_shape.empty()){
+              result.reshape({1});
```

Comment:
Just some nitpicking regardign formatting:

- 4 spaces indennt, space after if and squiggly bracket on newline:

```
if (result_shape.empty())
{
    result.reshape({1});
}
else
{
...
}
```